studio/components/layouts/ProjectLayout/LayoutHeader/NotificationsPopover/NotificationActions.tsx

```diff
@@ -1,39 +1,104 @@
-import { FC } from 'react'
+import { FC, useState } from 'react'
 import { useRouter } from 'next/router'
 import { Button } from '@supabase/ui'
+import { Action, ActionType } from '@supabase/shared-types/out/notifications'
 
 import { Project } from 'types'
+import { API_URL } from 'lib/constants'
+import { post, delete_, patch } from 'lib/common/fetch'
+import { useStore } from 'hooks/misc/useStore'
+import ConfirmModal from 'components/ui/Dialogs/ConfirmDialog'
 
 interface Props {
   project: Project
-  availableActions: any[]
+  availableActions: Action[]
   onSelectRestartProject: () => void
 }
 
 const NotificationActions: FC<Props> = ({ project, availableActions, onSelectRestartProject }) => {
   const router = useRouter()
+  const { app } = useStore()
+  const [showModal, setShowModal] = useState<boolean>(false)
 
   const onSelectUpgradeProject = () => {
     return router.push(`/project/${project.ref}/settings/billing/update/pro`)
   }
 
+  // TODO: allow configuring other routes
+  const onSelectMigrateProject = () => {
+    post(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+  const onSelectRollbackProject = () => {
+    delete_(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+  const onSelectFinaliseProject = () => {
+    patch(`${API_URL}/platform/database/${project.ref}/owner-reassign`, {})
+      // Refresh database connection string
+      .then((_) => app.projects.fetchDetail(project.ref))
+      .catch((err) => console.info(err))
+  }
+
   return (
     <div className="space-y-2">
-      {availableActions.map((action: any) => {
-        if (action.action_type === 'project.upgrade') {
+      {availableActions.map((action) => {
+        if (action.action_type === ActionType.UpgradeProjectToPro) {
           return (
             <Button key={action.action_type} type="default" onClick={onSelectUpgradeProject}>
               Upgrade project
             </Button>
           )
-        } else if (action.action_type === 'postgresql.restart') {
+        } else if (action.action_type === ActionType.SchedulePostgresRestart) {
           return (
             <Button key={action.action_type} type="default" onClick={onSelectRestartProject}>
               Restart project
             </Button>
           )
+        } else if (action.action_type === ActionType.MigratePostgresSchema) {
+          if (action.reason === 'finalise') {
+            return (
+              <Button
+                key={`${action.action_type}_${action.reason}`}
+                type="default"
+                onClick={() => setShowModal(true)}
+              >
+                Finalise
```

Comment:
nit: we use American English spelling; for consistency, use `finalize` rather than `finalise`